tests/unit/test_charm.py

```diff
@@ -134,7 +124,7 @@ def test_given_requirer_and_no_outstanding_certs_when_get_outstanding_certificat
         self.harness.add_relation("certificates", "requirer")
         patch_get_requirer_units_csrs_with_no_certs.return_value = []
         event = Mock()
-        self.harness.charm._on_get_outstanding_certificate_requests_action(event=event)
+        self.harness.charm._on_get_outstanding_certificate_requests_action(event=event)  # type: ignore[reportAttributeAccessIssue]
```

Comment:
Instead of adding all these ignores, I recommend using the `harness.run_action`.

Example 1:

```python
with self.assertRaises(ActionFailed) as e:
    self.harness.run_action("get-outstanding-certificate-requests")

self.assertEqual("No certificates relation has been created yet.", e.exception.message)
```

Example 2:

```python
action_output = self.harness.run_action("get-outstanding-certificate-requests")
self.assertEqual(
    json.dumps([vars(requirer_csr)]), action_output.results["result"]
)
```

Reply:
Done